System/Random.hs

```diff
@@ -543,87 +525,130 @@ class Monad m => MonadRandom (g :: * -> *) s m | m -> s where
     l32 <- uniformWord32 g
     h32 <- uniformWord32 g
     pure (unsafeShiftL (fromIntegral h32) 32 .|. fromIntegral l32)
-  uniformByteArray :: Int -> g s -> m ByteArray
-  default uniformByteArray :: PrimMonad m => Int -> g s -> m ByteArray
-  uniformByteArray = uniformByteArrayPrim
-  {-# INLINE uniformByteArray #-}
+  uniformShortByteString :: Int -> g s -> m ShortByteString
+  default uniformShortByteString :: MonadIO m => Int -> g s -> m ShortByteString
```

Comment:
The `MonadIO` constraint makes it difficult to create a `PrimMonad` based `MonadRandom` instance for `MWC.Gen`, for example:

```
/home/leo/Code/random/System/Random.hs:424: failure in expression `:{
 instance (s ~ PrimState m, PrimMonad m) => MonadRandom MWC.Gen s m where
     newtype Frozen MWC.Gen = Frozen { unFrozen :: MWC.Seed }
     thawGen = fmap MWC.restore unFrozen
     freezeGen = fmap Frozen . MWC.save
     uniformWord8 = MWC.uniform
     uniformWord16 = MWC.uniform
     uniformWord32 = MWC.uniform
     uniformWord64 = MWC.uniform
:}'
expected: 
 but got: 
          <interactive>:57:11: error:
              • Could not deduce (Control.Monad.IO.Class.MonadIO m)
                  arising from a use of ‘System.Random.$dmuniformShortByteString’
                from the context: (s ~ PrimState m, PrimMonad m)
                  bound by the instance declaration at <interactive>:57:11-67
                Possible fix:
                  add (Control.Monad.IO.Class.MonadIO m) to the context of
                    the instance declaration
              • In the expression:
                  System.Random.$dmuniformShortByteString @(MWC.Gen) @(s) @(m)
                In an equation for ‘uniformShortByteString’:
                    uniformShortByteString
                      = System.Random.$dmuniformShortByteString @(MWC.Gen) @(s) @(m)
                In the instance declaration for ‘MonadRandom MWC.Gen s m’
```

Reply:
That was expected, good thing there is doctest. I'll added a proper fix for this